src/Mvc/Mvc.Testing/src/DeferredHostBuilder.cs

```diff
@@ -38,8 +40,18 @@ public DeferredHostBuilder()
 
         public IHost Build()
         {
+            // Hosting configuration is being provided by args so that
+            // we can impact WebApplicationBuilder based applications.
+            var args = new List<string>();
+
+            // Transform the host configuration into command line arguments
+            foreach (var section in _hostConfiguration.GetChildren())
+            {
+                args.Add($"--{section.Key}={section.Value}");
```

Comment:
Does this need to worry about escaping anything if there's a funky key with a space in it or something?

Reply:
I think it needs to handle complex keys as well. I'm guessing that scenario might be broken as well.

Reply:
I don't know that I need to handle spaces since the args are already in their own array element. https://source.dot.net/#Microsoft.Extensions.Configuration.CommandLine/CommandLineConfigurationProvider.cs,99f51fe8a25ff886

Reply:
Cool - I thought I'd just mention it in case there was some nuance to it somewhere 👍